mitodl/ocw-hugo-themes
# Start linting typescript code

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/eslint-config/issues/15, https://github.com/mitodl/eslint-config/issues/14

#### What's this PR do?
Updates the linting configuration to support typescript. 

#### How should this be manually tested?
There should be no functional changes, although I did change a few small pieces of application code (e.g., adding missing dependencies to a useEffect hook). So, run any checks you like on OCW.
